File: app/src/main/java/com/alanpaine/sphtech/mvp/main/adapter/section/provider/SecondNodeProvider.java
package com.alanpaine.sphtech.mvp.main.adapter.section.provider;

import android.view.View;

import com.alanpaine.sphtech.R;
import com.alanpaine.sphtech.bean.section.ItemNode;
import com.chad.library.adapter.base.entity.node.BaseNode;
import com.chad.library.adapter.base.provider.BaseNodeProvider;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class SecondNodeProvider extends BaseNodeProvider {

    @Override
    public int getItemViewType() {
        return 1;
    }

    @Override
    public int getLayoutId() {
        return R.layout.item_main_yearly;
    }

    @Override
    public void convert(@NotNull BaseViewHolder helper, @Nullable BaseNode data) {
        if (data == null) {
            return;
        }
       // helper.setText(R.id.tv_year,"");
      //  helper.setText(R.id.tv_volume,"");
        /*ItemNode entity = (ItemNode) data;
        helper.setImageResource(R.id.iv, entity.getImg());
        helper.setText(R.id.tv, entity.getName());*/
    }

    @Override
    public void onClick(@NotNull BaseViewHolder helper, @NotNull View view, BaseNode data, int position) {
    }
}

File: app/src/main/java/com/alanpaine/sphtech/mvp/main/adapter/section/provider/SecondNodeProvider.kt
package com.alanpaine.sphtech.mvp.main.adapter.section.provider

import android.view.View
import com.alanpaine.sphtech.R
import com.alanpaine.sphtech.bean.section.ItemNode
import com.chad.library.adapter.base.entity.node.BaseNode
import com.chad.library.adapter.base.provider.BaseNodeProvider
import com.chad.library.adapter.base.viewholder.BaseViewHolder

class SecondNodeProvider : BaseNodeProvider(){
    override val itemViewType: Int
        get() = 1
    override val layoutId: Int
        get() = R.layout.item_main_yearly

    override fun convert(helper: BaseViewHolder, data: BaseNode) {
        val entity = data as ItemNode
        helper.setText(R.id.tv_year,"Year: ${entity.quarter}")
        helper.setText(R.id.tv_volume,"Volume: ${entity.volume_of_mobile_data}")
    }

    override fun onClick(helper: BaseViewHolder, view: View, data: BaseNode, position: Int) {
        super.onClick(helper, view, data, position)
    }

}